File: db/src/main/java/com/jakedelivery/db/userorder/UserOrderEntity.java
package com.jakedelivery.db.userorder;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.jakedelivery.db.BaseEntity;
import com.jakedelivery.db._common.constant.UserOrderStatus;
import com.jakedelivery.db.store.StoreEntity;
import com.jakedelivery.db.userordermenu.UserOrderMenuEntity;
import lombok.*;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@SuperBuilder
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "user_order")
@Entity
public class UserOrderEntity extends BaseEntity {
    @Column(nullable = false)
    private Long userId; // user table 1:n

    @JoinColumn(nullable = false, name="storeId")
    @ManyToOne
    private StoreEntity store;

    @Enumerated(EnumType.STRING)
    @Column(length = 50, nullable = false)
    private UserOrderStatus status;

    @Column(precision = 11, scale = 4, nullable = false)
    private BigDecimal amount;

    private LocalDateTime orderedAt;
    private LocalDateTime acceptedAt;
    private LocalDateTime cookingStartedAt;
    private LocalDateTime deliveryStartedAt;
    private LocalDateTime receivedAt;

    @JsonIgnore
    @ToString.Exclude
    @OneToMany(mappedBy = "userOrder")
    private List<UserOrderMenuEntity> userOrderMenuEntities;
}

File: db/src/main/kotlin/com/jakedelivery/db/userorder/UserOrderEntity.kt
package com.jakedelivery.db.userorder

import com.fasterxml.jackson.annotation.JsonIgnore
import com.jakedelivery.db._common.constant.UserOrderStatus
import com.jakedelivery.db.store.StoreEntity
import com.jakedelivery.db.userordermenu.UserOrderMenuEntity
import java.math.BigDecimal
import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "user_order")
class UserOrderEntity(
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @field:Column(nullable = false)
    var userId: Long? = null,

    @field:JoinColumn(nullable = false, name = "storeId")
    @field:ManyToOne
    var store: StoreEntity? = null,

    @field:Enumerated(EnumType.STRING)
    @field:Column(length = 50, nullable = false)
    var status: UserOrderStatus? = null,

    @field:Column(precision = 11, scale = 4, nullable = false)
    var amount: BigDecimal? = null,

    var orderedAt: LocalDateTime? = null,
    var acceptedAt: LocalDateTime? = null,
    var cookingStartedAt: LocalDateTime? = null,
    var deliveryStartedAt: LocalDateTime? = null,
    var receivedAt: LocalDateTime? = null,

    @field:JsonIgnore
    @field:OneToMany(mappedBy = "userOrder")
    var userOrderMenuEntities: MutableList<UserOrderMenuEntity>?= null
) {
    override fun toString(): String {
        return "UserOrderEntity(id=$id, userId=$userId, store=$store, status=$status, amount=$amount, orderedAt=$orderedAt, acceptedAt=$acceptedAt, cookingStartedAt=$cookingStartedAt, deliveryStartedAt=$deliveryStartedAt, receivedAt=$receivedAt)"
    }
}